File: src/server/InvestmentApp-Server/Models/Investors/InvestorProject.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;
using InvestmentApp.Models.Projects;

namespace InvestmentApp.Models.Investors;

public class InvestorProject
{
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public Guid Id { get; set; }

    public Investor Investor { get; set; }

    public Guid InvestorId { get; set; }

    public Project Project { get; set; }

    public Guid ProjectId { get; set; }

    public double MinIncomeRate { get; set; }

    public double MaxRiskRate { get; set; }
}

File: src/client/src/app/models/investor-project.ts
import { DataTableRow, Project } from ".";
import { Guid } from "guid-typescript";

// TODO: add investor
export interface InvestorProject {
    id: Guid
    minIncomeRate: number;
    maxRiskRate: number;
    project: Project;
    projectId: Guid;
}

export interface InvestorProjectRow extends DataTableRow {
    id: string,
    internalId: string;
    minIncomeRate: number;
    maxRiskRate: number;
    project: string;
    projectId: string;
}
